mmocr/datasets/transforms/ocr_transforms.py

```diff
@@ -613,3 +615,136 @@ def __repr__(self):
         repr_str += f'backend={self.backend}), '
         repr_str += f'interpolation={self.interpolation})'
         return repr_str
+
+
+@TRANSFORMS.register_module()
+class RemoveIgnored(BaseTransform):
+    """Removed ignored elements from the pipeline.
+
+    Required Keys:
+
+    - gt_ignored
+    - gt_polygons (optional)
+    - gt_bboxes (optional)
+    - gt_bboxes_labels (optional)
+    - gt_texts (optional)
+
+    Modified Keys:
+
+    - gt_ignored
+    - gt_polygons (optional)
+    - gt_bboxes (optional)
+    - gt_bboxes_labels (optional)
+    - gt_texts (optional)
+    """
+
+    def transform(self, results: Dict) -> Dict:
+        remove_inds = np.where(results['gt_ignored'])[0]
+        if len(remove_inds) == len(results['gt_ignored']):
+            return None
+        return remove_pipeline_elements(results, remove_inds)
+
+
+@TRANSFORMS.register_module()
+class FixInvalidPolygon(BaseTransform):
```

Comment:
It's maybe better to split into two `Transform`:
* RemovePolyPointLess: for removing the polygon whose length is less than `min_points`
* FixSelfCrossingOrSelfTouching(mode='fix/remove/ignore', remove_when_failed: bool, failed_replace_with_bbox: bool): for removing the polygon when it is self-crossing or self-torching

Reply:
I don't think it's needed, because all of our transforms and evaluation metrics are built upon the assumption that these polygons are valid in `shapely`, and all this transform does is to ensure that no invalid polygons will be in the pipeline. Even though we have split this into two transforms, they will always be used together in order to not produce errors in the upcoming `shapely` operations.

In other words, if users don't need to rely on `shapely` in their implementation, they will never need to call `FixInvalidPolygon`. As for now, we can just keep this transform simple for most users.